libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -515,22 +516,41 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync
     };
     
     // Create handler
-    SFSyncServerTargetCompleteBlock completeBlockCreate = ^(NSDictionary *d) {
+    SFSyncUpTargetCompleteBlock completeBlockCreate = ^(NSDictionary *d) {
         // Replace id with server id during create
         record[kId] = d[kSyncManagerLObjectId];
         completeBlockUpdate(d);
     };
     
-    SFSyncServerTargetCompleteBlock completeBlock;
+    
+    SFSyncUpTarget *target = (SFSyncUpTarget*) sync.target;
+    NSArray *fieldList = sync.options.fieldlist;
+    
+    // Getting type and id
+    NSString* objectType = [SFJsonUtils projectIntoJson:record path:kObjectTypeField];
+    NSString* objectId = record[kId];
+    
+    // Fields to save (in the case of create or update)
+    NSMutableDictionary* fields = [NSMutableDictionary dictionary];
```

Comment:
Part of me feel that sync options and sync target could be merged. With possibly the merge mode staying out.
Anyway for your app, are the methods current signature too restrictive?